libsolidity/interface/CompilerStack.cpp

```diff
@@ -782,6 +782,28 @@ void CompilerStack::link()
 	}
 }
 
+YulStack CompilerStack::loadGeneratedIR(std::string const& _ir) const
+{
+	YulStack stack(
+		m_evmVersion,
+		m_eofVersion,
+		YulStack::Language::StrictAssembly,
+		m_optimiserSettings,
+		m_debugInfoSelection,
+		this, // _soliditySourceProvider
+		m_objectOptimizer
+	);
+	bool yulAnalysisSuccessful = stack.parseAndAnalyze("", _ir);
+	solAssert(
+		yulAnalysisSuccessful,
+		_ir + "\n\n"
+		"Invalid IR generated:\n" +
+		SourceReferenceFormatter::formatErrorInformation(stack.errors(), stack) + "\n"
+	);
```

Comment:
Since you're already giving a detailed explanation with the assert, perhaps it is also a good fit for the error reporter instead of asserting? Although then, I assume, you would also need to add some `solThrow` afterwards...

Reply:
It's an assert on purpose. The code we're parsing comes from the codegen and is not supposed to have any errors (or even warnings). If there are any, it's a bug, not something that's supposed to be reported to the user. The user does is not even necessarily aware of the IR and including errors that point at locations that are not located their Solidity sources would be quite confusing.

Reply:
Fair enough in that case.